src/decode.rs

```diff
@@ -4325,18 +4324,11 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
         }
 
         let lowest_pixel_mem_sz = frame_hdr.tiling.cols * f.sbh;
-        if lowest_pixel_mem_sz != f.tile_thread.lowest_pixel_mem_sz {
-            free(f.tile_thread.lowest_pixel_mem as *mut c_void);
-            f.tile_thread.lowest_pixel_mem =
-                malloc(lowest_pixel_mem_sz as usize * ::core::mem::size_of::<[[c_int; 2]; 7]>())
-                    as *mut [[c_int; 2]; 7];
-            if f.tile_thread.lowest_pixel_mem.is_null() {
-                f.tile_thread.lowest_pixel_mem_sz = 0;
-                return Err(ENOMEM);
-            }
-            f.tile_thread.lowest_pixel_mem_sz = lowest_pixel_mem_sz;
-        }
-        let mut lowest_pixel_ptr = f.tile_thread.lowest_pixel_mem;
+        // TODO: Fallible allocation
+        f.lowest_pixel_mem
+            .resize(lowest_pixel_mem_sz as usize, Default::default());
```

Comment:
```suggestion
            .resize_with(lowest_pixel_mem_sz as usize, Default::default);
```

Reply:
Why though? What's the difference between these two? Data is `Copy` so the end result is the same.

Reply:
The docs recommend it for `Default::default` and it optimizes a little bit better, since `.resize` does a write for `n - 1`, cloning them, and then writes the last one without the clone.  `.resize_with` just write all `n` at once, so the codegen is better, and the source is a bit cleaner.

Reply:
Noted. I'll use `resize_with` going forward.